drivers/CoreLED/include/CoreLED.h

```diff
@@ -5,39 +5,110 @@
 #ifndef _LEKA_OS_DRIVER_CORE_LED_H_
 #define _LEKA_OS_DRIVER_CORE_LED_H_
 
+#include <algorithm>
+
 #include "RGB.h"
 #include "interface/drivers/LED.h"
 #include "interface/drivers/SPI.h"
 
 namespace leka {
 
+template <size_t NumberOfLeds>
 class CoreLED : public interface::LED
 {
   public:
-	explicit CoreLED(interface::SPI &spi, int size) : _spi {spi}, _size(size) {};
+	explicit CoreLED(interface::SPI &spi) : _spi {spi} {};
+
+	void setColor(const RGB &color) override { std::fill(_colors.begin(), _colors.end(), color); }
+
+	void setColorAtIndex(const int index, const RGB &color) override
+	{
+		if (index >= NumberOfLeds) {
+			return;
+		}
+		_colors.at(index) = color;
+	}
+
+	void setColorWithArray(const std::span<const RGB> color) override
+	{
+		std::copy(color.begin(), color.end(), _colors.begin());
+	}
+
+	void setColorRange(int start, int end, const RGB &color) override
+	{
+		if (start >= NumberOfLeds && end >= NumberOfLeds) {
+			return;
+		}
+
+		if (start > end) {
+			std::swap(start, end);
+		}
+
+		if (end >= NumberOfLeds) {
+			end = NumberOfLeds - 1;
+		}
+
+		std::fill(_colors.begin() + start, _colors.begin() + end + 1, color);
+	}
+
+	void showColor() override
+	{
+		sendAndDisplay(_colors);
 
-	void setColor(RGB color) override;
-	void showColor() override;
-	void hideColor() override;
+		auto led_is_not_black		= [](const auto &c) { return c != RGB::black; };
+		auto all_leds_are_not_black = std::all_of(_colors.begin(), _colors.end(), led_is_not_black);
 
-	auto isOn() -> bool override;
+		_is_color_shown = all_leds_are_not_black;
+	}
+
+	void hideColor() override
+	{
+		if (!isOn()) {
+			return;
+		}
+
+		constexpr auto black = [] {
+			auto colors = std::array<RGB, NumberOfLeds> {};
+			std::fill(colors.begin(), colors.end(), RGB::black);
+			return colors;
+		}();
+
+		sendAndDisplay(black);
+		_is_color_shown = false;
```

Comment:
Pourquoi ne pas avoir fait
```suggestion	
		setColor(RGB::black);
		showColor();
```
C'est à propos des termes utilisés?

Reply:
parce que setColor il set la color donc on perd l'ancienne. là on affiche juste du noir, on peut ensuite la showColor pour la remontrer